src/DAMA/MenuBundle/Resources/config/services.yml

```diff
@@ -1,7 +1,10 @@
 services:
   dama_menu.twig.menu_extension:
     class: DAMA\MenuBundle\Twig\MenuExtension
-    arguments: ["@service_container"]
+    arguments:
+      - "@twig"
```

Comment:
also wondering: does this not produce a circular dependency somehow?

Reply:
Well, it works :D I guess not :)

Reply:
:smile:

Reply:
Just curious if pulling twig from container is not the same? :)

Reply:
not sure I follow? the `twig` service must already be instantiated anyway if we we using any of the methods provided by `DAMA\MenuBundle\Twig\MenuExtension` as we are rendering a template then :blush:

Reply:
I mean is there a difference in pulling twig from injected container or inject a twig? Aren't they instantiated only once?

Reply:
Ah thats what you mean. For `@twig` it does not make any difference.

But for `@dama_menu.menu_factory` and `@dama_menu.menu_config_provider` it does make a difference as the methods of this extension might not be needed for a lot of requests so often we instantiate them for no reason.

Better be lazy here :blush:

Reply:
I always thought that Public means that they are not lazy :)